### PR TITLE
[AIP-6492] Supporting JSONType as param

### PR DESCRIPTION
**Pull request by cavandervoort**

### CODE REVIEW

File: hello_jsontype.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                            default="[1, 2, 3]")
+    str_to_dict = Parameter("str_to_dict",
+                            type=JSONType,
+                            default='{"one": 1, "two": 2, "three": 3}')
+    list_param = Parameter("list_param",
+                           type=JSONType,
+                           default="[1, 2, 3]")
+    dict_param = Parameter("dict_param",
+                           type=JSONType,
+                           default='{"one": 1, "two": 2, "nine": 9}')
```
Comment (cloudw) [inline hello_jsontype.py:21]: Nice that you are testing all the cases.

File: hello_jsontype.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    def hello(self):
+        """
+        A step to print out JSONType objects.
+
+        """
+        print("Ready to begin testing JSONType object params")
+        print(f"str_to_list - type: {type(self.str_to_list)}, data: {self.str_to_list}")
+        print(f"str_to_dict - type: {type(self.str_to_dict)}, data: {self.str_to_dict}")
+        print(f"list_param - type: {type(self.list_param)}, data: {self.list_param}")
+        print(f"dict_param - type: {type(self.dict_param)}, data: {self.dict_param}")
```
Comment (cloudw) [inline hello_jsontype.py:43]: I suggest asserting that all of these parameters have NoneProd data, and actually assert that the values equal to what you pass in from `test_run_jsontype.py`. This way you will know that the test fails if corresponding metaflow run fails.

**Review by cloudw — APPROVED** [review]
Good test cases and clean handling of json conversion with `try ... except`. One minor suggestions for the test case. 

Mentioning two more things, not necessarily part of this PR, that I'd like to have you get used to:
- Can you write a bit more in the PR description? This PR is relatively short with self-descriptive title but it's mostly to get started on the habit.  
  - This is more important in the future people reviewing your PR doesn't have the same context as I do. It's also a nice record of why the changes are made if anyone (sometimes yourself) wonders months later when debugging.
  - The description can contain a summary of what's being changed - especially if there are multiple commits or lots of code changes.
  - More importantly describe why a change is needed since this part of info can't be found in code.
- Use formatter (should take you only 5 min)
  - In Metaflow we happen to have `black` setup for pre-commit: https://github.com/zillow/metaflow/blob/zillow/argo/.pre-commit-config.yaml. To utilize it, follow step 1 and 3 of this quick start guide https://pre-commit.com/#quick-start.
  - With this setup, all your future commit in the local git repo will be automatically formatted.

File: hello_jsontype.py (hunk 1 above)
Comment (cavandervoort) [reply hello_jsontype.py:21]: Thanks

File: hello_jsontype.py (hunk 2 above)
Comment (cavandervoort) [reply hello_jsontype.py:43]: I will do this for JSONType tests and all other tests in [AIP-6471]. I pasted this feedback in a comment on that task.

**Comment by cavandervoort** [comment]
> Good test cases and clean handling of json conversion with `try ... except`. One minor suggestions for the test case.
> 
> Mentioning two more things, not necessarily part of this PR, that I'd like to have you get used to:
> 
> * Can you write a bit more in the PR description? This PR is relatively short with self-descriptive title but it's mostly to get started on the habit.
>   
>   * This is more important in the future people reviewing your PR doesn't have the same context as I do. It's also a nice record of why the changes are made if anyone (sometimes yourself) wonders months later when debugging.
>   * The description can contain a summary of what's being changed - especially if there are multiple commits or lots of code changes.
>   * More importantly describe why a change is needed since this part of info can't be found in code.
> * Use formatter (should take you only 5 min)
>   
>   * In Metaflow we happen to have `black` setup for pre-commit: https://github.com/zillow/metaflow/blob/zillow/argo/.pre-commit-config.yaml. To utilize it, follow step 1 and 3 of this quick start guide https://pre-commit.com/#quick-start.
>   * With this setup, all your future commit in the local git repo will be automatically formatted.

**Comment by cavandervoort** [comment]
Re your comment above re PR description and using black for formatting, I took note of it and will follow your advice in future PRs

**Pull request merged**